htlcswitch/switch.go

```diff
@@ -985,6 +985,35 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			return s.failAddPacket(packet, failure)
 		}
 
+		// Check if the incoming channel link has been Disabled by the user.
+		// If so, reject the HTLC.
+		s.indexMtx.RLock()
+		incomingLink, err := s.getLinkByShortID(packet.incomingChanID)
+		if err != nil {
+			s.indexMtx.RUnlock()
```

Comment:
Move `RUnlock()` directly after `getLinkByShortID()`? Then it only needs to be called once.